src/vs/editor/contrib/parameterHints/browser/parameterHintsWidget.ts

```diff
@@ -272,12 +275,30 @@ export class ParameterHintsWidget extends Disposable implements IContentWidget {
 	}
 
 	private renderMarkdownDocs(markdown: IMarkdownString | undefined): IMarkdownRenderResult {
+		const stopWatch = new StopWatch();
 		const renderedContents = this.renderDisposeables.add(this.markdownRenderer.render(markdown, {
 			asyncRenderCallback: () => {
 				this.domNodes?.scrollbar.scanDomNode();
 			}
 		}));
 		renderedContents.element.classList.add('markdown-docs');
+
+		type RenderMarkdownPerformanceClassification = {
+			owner: 'donjayamanne';
+			comment: 'Measure the time taken to render markdown for parameter hints';
+			renderDuration: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; comment: 'Time in ms to render the markdown' };
+		};
+
+		type RenderMarkdownPerformanceEvent = {
+			renderDuration: number;
+		};
+		const renderDuration = stopWatch.elapsed();
+		if (renderDuration > 300) {
+			this.telemetryService.publicLog2<RenderMarkdownPerformanceEvent, RenderMarkdownPerformanceClassification>('parameterHints.parseMarkdown', {
+				renderDuration: stopWatch.elapsed()
```

Comment:
[nit] use renderDuration again here

Reply:
Absolutely, that was the original plan, but didn't complete that.